contracts/token/BiconomyTokenPaymaster.sol

```diff
@@ -578,31 +592,31 @@ contract BiconomyTokenPaymaster is
         // This model assumes irrespective of priceSource exchangeRate is always sent from outside
         // for below checks you would either need maxCost or some exchangeRate
 
-        // review: can add some checks here on calculated value, fee cap, exchange rate
         uint256 tokenRequiredPreFund = ((requiredPreFund + costOfPost) *
             exchangeRate) / 10 ** 18;
         require(
             tokenRequiredPreFund != 0,
             "BTPM: calculated token charge invalid"
         );
-        require(
-            fee <= (tokenRequiredPreFund * 20) / 100,
-            "BTPM: fee markup too high"
-        );
+
+        // check for caps after applying markup against without markup
+        require(priceMarkup <= 2e6, "BTPM: price markup percentage too high");
```

Comment:
Require priceMarkup to be greater than 0 also.

Reply:
greater than 0 or 1e6?
do we want to allow fee rebates? 

1e6 means 1x (no markup!)

Reply:
i am keeping >=1e6 <=2e6